server/src/main/antlr3/org/apache/iotdb/db/sql/parse/TqlParser.g

```diff
@@ -0,0 +1,726 @@
+
+dataType
+    : K_INT32 | K_INT64 | K_FLOAT | K_DOUBLE | K_BOOLEAN | K_TEXT
+    ;
+
+setClause
+    : K_SET setCol (COMMA setCol)*
+    -> setCol+
+    ;
+
+setCol
+    : suffixPath OPERATOR_EQ constant
+    -> ^(TOK_VALUE suffixPath constant)
+    ;
+
+deleteTimeseries
+    : K_DELETE K_TIMESERIES prefixPath (COMMA prefixPath)*
+    -> ^(TOK_DELETE ^(TOK_TIMESERIES prefixPath+))
+    ;
+
+setStorageGroup
+    : K_SET K_STORAGE K_GROUP K_TO prefixPath
+    -> ^(TOK_SET ^(TOK_STORAGEGROUP prefixPath))
+    ;
+
+deleteStorageGroup
+    : K_DELETE K_STORAGE K_GROUP prefixPath
+    ;
+
+createProperty
+    : K_CREATE K_PROPERTY ID
+    -> ^(TOK_CREATE ^(TOK_PROPERTY ID))
+    ;
+
+addLabel
+    : K_ADD K_LABEL label=ID K_TO K_PROPERTY propertyName=ID
+    -> ^(TOK_ADD ^(TOK_LABEL $label) ^(TOK_PROPERTY  $propertyName))
+    ;
+
+deleteLabel
+    : K_DELETE K_LABEL label=ID K_FROM K_PROPERTY propertyName=ID
+    -> ^(TOK_DELETE ^(TOK_LABEL $label) ^(TOK_PROPERTY $propertyName))
+    ;
+
+linkPath
+    : K_LINK prefixPath K_TO propertyLabelPair
+    -> ^(TOK_LINK prefixPath propertyLabelPair)
+    ;
+
+propertyLabelPair
+    : propertyName=ID DOT labelName=ID
+    -> ^(TOK_LABEL $labelName) ^(TOK_PROPERTY $propertyName)
+    ;
+
+unlinkPath
+    :K_UNLINK prefixPath K_FROM propertyLabelPair
+    -> ^(TOK_UNLINK prefixPath  propertyLabelPair)
+    ;
+
+showMetadata
+    : K_SHOW K_METADATA
+    -> ^(TOK_SHOW_METADATA)
+    ;
+
+describePath
+    : K_DESCRIBE prefixPath
+    -> ^(TOK_DESCRIBE prefixPath)
+    ;
+
+createIndex
+    : K_CREATE K_INDEX K_ON timeseriesPath K_USING function=ID indexWithClause? whereClause?
+    -> ^(TOK_CREATE ^(TOK_INDEX timeseriesPath ^(TOK_FUNCTION $function indexWithClause? whereClause?)))
+    ;
+
+indexWithClause
+    : K_WITH indexValue (COMMA indexValue)?
+    -> ^(TOK_WITH indexValue+)
+    ;
+
+indexValue
+    : ID OPERATOR_EQ INT
+    -> ^(TOK_INDEX_KV ID INT)
+    ;
+
+dropIndex
+    : K_DROP K_INDEX function=ID K_ON timeseriesPath
+    -> ^(TOK_DROP ^(TOK_INDEX timeseriesPath ^(TOK_FUNCTION $function)))
+    ;
+
+mergeStatement
+    : K_MERGE
+    -> ^(TOK_MERGE)
+    ;
+
+listStatement
+    : K_LIST K_USER -> ^(TOK_LIST TOK_USER)
+    | K_LIST K_ROLE -> ^(TOK_LIST TOK_ROLE)
+    | K_LIST K_PRIVILEGES K_USER username = ID K_ON prefixPath -> ^(TOK_LIST TOK_PRIVILEGES ^(TOK_USER $username) prefixPath)
+    | K_LIST K_PRIVILEGES K_ROLE roleName = ID K_ON prefixPath -> ^(TOK_LIST TOK_PRIVILEGES ^(TOK_ROLE $roleName) prefixPath)
+    | K_LIST K_USER K_PRIVILEGES username = ID -> ^(TOK_LIST TOK_PRIVILEGES TOK_ALL ^(TOK_USER $username))
+    | K_LIST K_ROLE K_PRIVILEGES roleName = ID -> ^(TOK_LIST TOK_PRIVILEGES TOK_ALL ^(TOK_ROLE $roleName))
+    | K_LIST K_ALL K_ROLE K_OF K_USER username = ID -> ^(TOK_LIST TOK_ROLE TOK_ALL ^(TOK_USER $username))
+    | K_LIST K_ALL K_USER K_OF K_ROLE roleName = ID -> ^(TOK_LIST TOK_USER TOK_ALL ^(TOK_ROLE $roleName))
+    ;
+
+quitStatement
+    : K_QUIT
+    -> ^(TOK_QUIT)
+    ;
```

Comment:
I wonder whether the "quit" statement has a chance to reach the server.

Reply:
Actually, even if the statement is a quit statement, no logical plan will be generated. Do you suggest removing the grammar definition here?

Reply:
Yes, redundant things should always be removed.

Reply:
These codes are removed. 
Thanks for your suggestion!